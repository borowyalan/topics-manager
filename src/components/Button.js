import React, { Component } from 'react';
import styled from 'styled-components';

class Button extends Component {

    render() {
        return (
            this.props.type === 'primary' ?
                <PrimaryButton onClick={this.props.onClick}>{this.props.children}</PrimaryButton>
                : <SecondaryButton onClick={this.props.onClick}>{this.props.children}</SecondaryButton>
        );
    }

}

const BasicButton = styled.button`
    border: none;
`;

const PrimaryButton = BasicButton.extend`
  background-color: red;
  color: #fff;
`;

const SecondaryButton = BasicButton.extend`
  backround-color: yellow;
  color: #000;
`;



export default Button;