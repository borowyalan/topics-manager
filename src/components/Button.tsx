import React, { SFC } from 'react';
import styled from 'styled-components';
import Colors from '../styles/colors';

export enum ButtonType {
    Primary = 'primary',
    Secondary = 'secondary',
}

export interface IButtonProps {
    kind?: ButtonType,
}

type Properties = IButtonProps & React.AllHTMLAttributes<HTMLButtonElement>

const Button: SFC<Properties> = (props) => {
    switch (props.kind) {
        case ButtonType.Primary:
            return <PrimaryButton {...props} onClick={props.onClick}>
                {props.children}
            </PrimaryButton>;
        case ButtonType.Secondary:
            return <SecondaryButton {...props} onClick={props.onClick}>
                {props.children}
            </SecondaryButton>;
        default:
            return null;
    }
};

const BasicButton = styled.button`
    border: none;
    padding: 15px;
    font-size: 14px;
    min-width: 180px;
    text-transform: uppercase;
    &:hover {
        box-shadow: 3px 7px 20px 0 rgba(0, 0, 0, 0.2);
        cursor: pointer;
    }
`;

const PrimaryButton = BasicButton.extend`
    background-color: ${Colors.red};
    color: ${Colors.white};
`;

const SecondaryButton = BasicButton.extend`
    background-color: ${Colors.yellow};
    color: ${Colors.black};
`;

export default Button;
