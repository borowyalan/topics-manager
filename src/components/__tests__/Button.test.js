import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import Button from '../Button';
import Adapter from 'enzyme-adapter-react-16';
import {configure} from 'enzyme/build/index';

configure({ adapter: new Adapter() });

describe('<Button> component', () => {
    let wrapper = shallow(<Button type='primary'/>);

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should have correct props', () => {
        expect(wrapper.props().type).toBe('primary');
    });
});
