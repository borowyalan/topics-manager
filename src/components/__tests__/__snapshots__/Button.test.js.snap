// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> component should render 1`] = `
<Button__BasicButton
  type="primary"
/>
`;
